enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/tx/ChunkedTransactionStream.java

```diff
@@ -96,6 +98,11 @@ else if ( txCursor.next() )
 
             CommittedTransactionRepresentation tx = txCursor.get();
             lastTxId = tx.getCommitEntry().getTxId();
+            if ( lastTxId != expectedTxId )
+            {
+                throw new IllegalStateException( "Transaction cursor out of order." );
```

Comment:
May be worth providing the expected and actual values of the transaction cursor to help with debugging from the logs?

Reply:
Yes, will fix!